Update and fix docs (namespaces, consistency)

dbort opened the pull request:
Summary:
Audit all instances of `\bexec_aten::` and `\btorch::` under `docs/`, updating where appropriate.

The only remaining `torch::` instances are for kernels, which I didn't get a chance to migrate before v0.4.0.

Also:
- Update the LLM Manual code to be consistent between the doc and main.cpp
- Fix some LLM Manual issues: point to the latest release, and "main.cpp" instead of "main.h"

Differential Revision: D64152344

malfet commented on examples/llm_manual/main.cpp:
```diff
@@ -90,7 +90,7 @@ std::string generate(
 int main() {
   // Set up the prompt. This provides the seed text for the model to elaborate.
-  std::cout << "Prompt: ";
+  std::cout << "Enter model prompt: ";
```
This change seems unrelated to PR title

malfet commented on docs/source/running-a-model-cpp-tutorial.md:
```diff
@@ -75,12 +86,12 @@ for (size_t id = 0; id < num_memory_planned_buffers; ++id) {
 HierarchicalAllocator planned_memory(
     {planned_arenas.data(), planned_arenas.size()});
 
-// Version of MemoryAllocator that uses malloc to handle allocations
-// rather then a fixed buffer.
-util::MallocMemoryAllocator method_allocator;
+// Version of MemoryAllocator that uses malloc to handle allocations rather then
+// a fixed buffer.
+MallocMemoryAllocator method_allocator;
 
-// Assemble all of the allocators into the MemoryManager that the Executor
-// will use.
+// Assemble all of the allocators into the MemoryManager that the Executor will
+// use.
```
Do not seem related to the PR title

malfet commented on docs/source/running-a-model-cpp-tutorial.md:
```diff
@@ -75,12 +86,12 @@ for (size_t id = 0; id < num_memory_planned_buffers; ++id) {
-// Version of MemoryAllocator that uses malloc to handle allocations
-// rather then a fixed buffer.
-util::MallocMemoryAllocator method_allocator;
+// Version of MemoryAllocator that uses malloc to handle allocations rather then
+// a fixed buffer.
```
Is there a linter that prefers one layout vs another?
line 89 now longer than 80 characters, isn't it?

malfet commented on docs/source/running-a-model-cpp-tutorial.md:
```diff
@@ -64,7 +74,8 @@ std::vector<Span<uint8_t>> planned_arenas; // Passed to the allocator
 size_t num_memory_planned_buffers = method_meta->num_memory_planned_buffers();
 
-// It is possible to have multiple layers in our memory hierarchy; for example, SRAM and DRAM.
+// It is possible to have multiple layers in our memory hierarchy; for example,
+// SRAM and DRAM.
```
Same as below, please don't change formatting (or mention in PR description which new linter rule mandates one to do so)

malfet commented on docs/source/running-a-model-cpp-tutorial.md:
```diff
@@ -48,14 +59,13 @@ One of the principles of ExecuTorch is giving users control over where the memor
-// information about a specific method. In this case we are looking to
-// get the required size of the memory planned buffers for the method
-// "forward".
+// information about a specific method. In this case we are looking to get the
+// required size of the memory planned buffers for the method "forward".
```
Unrelated to PR title

malfet commented on docs/source/llm/getting-started.md:
```diff
@@ -309,28 +309,28 @@ penalties for repeated tokens, and biases to prioritize or de-prioritize specifi
-    const auto max_output_tokens = 30;
-    std::cout << prompt;
-    generate(model, prompt, tokenizer, sampler, max_input_tokens, max_output_tokens);
+  // Set up the prompt. This provides the seed text for the model to elaborate.
+  std::cout << "Enter model prompt: ";
+  std::string prompt;
+  std::getline(std::cin, prompt);
+
+  // The tokenizer is used to convert between tokens (used by the model) and
+  // human-readable strings.
+  BasicTokenizer tokenizer("vocab.json");
+
+  // The sampler is used to sample the next token from the logits.
+  BasicSampler sampler = BasicSampler();
+
+  // Load the exported nanoGPT program, which was generated via the previous
+  // steps.
+  Module model("nanogpt.pte", Module::LoadMode::MmapUseMlockIgnoreErrors);
+
+  const auto max_input_tokens = 1024;
+  const auto max_output_tokens = 30;
+  std::cout << prompt;
+  generate(
+      model, prompt, tokenizer, sampler, max_input_tokens, max_output_tokens);
```
Unrelated to PR title

malfet reviewed (approved):
This PR contains a lots of changes unrelated to the title. If they must be landed together, please update the description

dbort commented:
@malfet I mention it in the summary as "Also update the LLM Manual code to be consistent between the doc and main.cpp." Basically I had to update the namespaces in both main.cpp and in the doc that includes snippets of main.cpp, but noticed that they're different. I reformatted them and made them consistent as part of this.

dbort commented:
Updated title/description because the scope of this creeped further

dbort committed:
Update and fix docs (namespaces, consistency) (#6084)

Summary:

Audit all instances of `\bexec_aten::` and `\btorch::` under `docs/`, updating where appropriate.

The only remaining `torch::` instances are for kernels, which I didn't get a chance to migrate before v0.4.0.

Also:
- Update the LLM Manual code to be consistent between the doc and main.cpp
- Fix some LLM Manual issues: point to the latest release, and "main.cpp" instead of "main.h"

Reviewed By: mergennachin

Differential Revision: D64152344

dbort commented:
@pytorchbot cherry-pick --onto release/0.4 -c docs
